libcxx/test/libcxx/containers/sequences/forwardlist/bool-conversion.pass.cpp

```diff
@@ -0,0 +1,38 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++20
+
+// <forward_list>
+
+// This test shows the effect of implementing `LWG4135`, before it this code
+// shouldn't compile with a long error message, as the predicate is not bool.
+// `LWG4135` suggests that std::erase explicitly specifying the lambda's
+// return type as bool.
```

Comment:
```suggestion
// This test shows the effect of implementing `LWG4135`, before it this code
// was ill-formed, as the predicate is not bool.
// `LWG4135` suggests that std::erase explicitly specifying the lambda's
// return type as bool.
```
A minor change to describe the issue in language of the Standard.
Especially the `a long error message` depends on the compiler used.